cmd/zfs/zfs_main.c

```diff
@@ -785,8 +785,7 @@ zfs_do_clone(int argc, char **argv)
 	return (!!ret);
 
 usage:
-	if (zhp)
-		zfs_close(zhp);
+	ASSERT(zhp != NULL);
```

Comment:
You mean `ASSERT3P(zhp, ==, NULL);` here since `zfs_open()` must never have been called.

Reply:
Next mistake on these macros and I owe you a beer.